app/app_main.c

```diff
@@ -1294,9 +1294,21 @@ static int enable_kdf(ACVP_CTX *ctx) {
     ACVP_RESULT rv = ACVP_SUCCESS;
     int i, flags = 0;
 
-    /*
-     * Enable KDF-135
-     */
+
+    rv = acvp_cap_kdf_tls12_enable(ctx, &app_kdf_tls12_handler);
+    CHECK_ENABLE_CAP_RV(rv);
+    rv = acvp_cap_set_prereq(ctx, ACVP_KDF_TLS12, ACVP_PREREQ_SHA, value);
+    CHECK_ENABLE_CAP_RV(rv);
+    rv = acvp_cap_set_prereq(ctx, ACVP_KDF_TLS12, ACVP_PREREQ_HMAC, value);
+    CHECK_ENABLE_CAP_RV(rv);
+    rv = acvp_cap_kdf_tls12_set_parm(ctx, ACVP_KDF_TLS12_HASH_ALG, ACVP_SHA256);
+    CHECK_ENABLE_CAP_RV(rv);
+    rv = acvp_cap_kdf_tls12_set_parm(ctx, ACVP_KDF_TLS12_HASH_ALG, ACVP_SHA384);
+    CHECK_ENABLE_CAP_RV(rv);
+    rv = acvp_cap_kdf_tls12_set_parm(ctx, ACVP_KDF_TLS12_HASH_ALG, ACVP_SHA512);
+    CHECK_ENABLE_CAP_RV(rv);
+
+#if 0 //replaced by KDF TLS12. Testing support will be removed from NIST server at the beginning of 2022.
```

Comment:
Is the new kdf already available on the production server ?

Reply:
Yes! according to the email I received about it

Reply:
from /acvp/v1/algorithms endpoint on prod:

            {
                "id": 176,
                "name": "TLS-v1.2",
                "mode": "KDF",
                "revision": "RFC7627"
            }